action.yml

```diff
@@ -60,6 +60,16 @@ runs:
         path: ${{ steps.poetry-venvs.outputs.dir }}
         key: poetry-venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
 
+    - name: Check that the poetry lockfile is up to date
+      # This is rather hacky. We look for the warning message in poetry's
+      # output. We really want to use `poetry lock --check`, but that is only
+      # available in poetry 1.2.
+      # https://github.com/python-poetry/poetry/issues/1406
+      run: >-
+        poetry export --without-hashes | (! grep "The lock file is not up to date") ||
+        (echo pyproject.toml was updated without running \`poetry lock --no-update\`. && false)
```

Comment:
If you were feeling particularly fancy, you could make this set an error annotation on poetry.lock: see https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-error-message

(But I don't think we _need_ to be fancy)

Reply:
Ooh, I didn't know you could do that with GHA. I'll leave the fanciness alone for now.